SWIG 4.4 requires different return value from `%init` fragments (mpi4py.i)

The context from which the SWIG generated wrapper calls `%init` fragments such as
https://github.com/mpi4py/mpi4py/blob/master/src/mpi4py/include/mpi4py/mpi4py.i#L13
has changed from a `PyObject *` returning-function to one that returns an int error code.

Right now that only leads to a warning but incorrect behavior in the initialization error case (which should rarely or never happen):
```
FSDataManager/include/FSDataManager.swigwrap_17.cxx:139966:35: warning: implicit conversion of NULL constant to 'int' [-Wnull-conversion]
 139966 |   if (import_mpi4py() < 0) return NULL; 
        |                            ~~~~~~ ^~~~
        |                                   0
1 warning generated.
```

Nonetheless, it would be good to fix this.

A relevant issue in SWIG with some guidance on how to `#ifdef` around this can be seen here: https://github.com/swig/swig/issues/3262

Personally, I'm doing
```
#if SWIG_VERSION >= 0x040400 // SWIG >= 4.4.0
  return -1;
#else
  return NULL;
#endif
```
in my wrapper(s).

--- a/src/mpi4py/include/mpi4py/mpi4py.i
+++ b/src/mpi4py/include/mpi4py/mpi4py.i
@@ -10,7 +10,11 @@
 /* ---------------------------------------------------------------- */
 
 %header %{ #include <mpi4py/mpi4py.h> %}
-%init   %{ if (import_mpi4py() < 0) return NULL; %}
+#if SWIG_VERSION >= 0x040400
+%init %{ if (import_mpi4py() < 0) return -1; %}
+#else
+%init %{ if (import_mpi4py() < 0) return NULL; %}
+#endif
 
 /* ---------------------------------------------------------------- */
 

